src/task/models/mergeTilesTaskBuilder.ts

```diff
@@ -112,25 +112,28 @@ export class MergeTilesTaskBuilder {
     const logger = this.logger.child({ logContext: { ...this.logContext, function: this.createPartLayers.name } });
     logger.debug({ msg: 'creating part layers', metadata: { partData, inputFiles, numberOfFiles: inputFiles.fileNames.length } });
     return inputFiles.fileNames.map<ILayerMergeData>((fileName) => {
-      const filePath = join(inputFiles.originDirectory, fileName);
+      const tilesPath = join(inputFiles.originDirectory, fileName);
       const footprint = partData.geometry;
       if (!footprint) {
-        logger.error({ msg: 'Part does not have a geometry', metadata: { partData } });
-        throw new Error('Part does not have a geometry');
+        const errorMsg = 'Part does not have a geometry';
+        logger.error({ msg: errorMsg, metadata: { partData } });
+        throw new Error(errorMsg);
       }
       const extent: BBox = bbox(footprint);
+      const maxZoom = degreesPerPixelToZoomLevel(partData.resolutionDegree ?? 0);
```

Comment:
need a fix after mc-models-types will be fixed, partData.resolutionDegree is a required field

Reply:
fixed